src/components/shared/DonationSummary.vue

```diff
@@ -9,60 +9,46 @@
-import PaymentSummaryUnset from '@/components/shared/payment_summary/PaymentSummaryUnset.vue';
-import PaymentSummaryCompany from '@/components/shared/payment_summary/PaymentSummaryCompany.vue';
-import PaymentSummaryEmail from '@/components/shared/payment_summary/PaymentSummaryEmail.vue';
-import PaymentSummaryPrivate from '@/components/shared/payment_summary/PaymentSummaryPrivate.vue';
+<script setup lang="ts">
+import { AddressTypeModel, addressTypeName } from '@src/view_models/AddressTypeModel';
+import PaymentSummaryAnonymous from '@src/components/shared/payment_summary/PaymentSummaryAnonymous.vue';
+import PaymentSummaryCompany from '@src/components/shared/payment_summary/PaymentSummaryCompany.vue';
+import PaymentSummaryEmail from '@src/components/shared/payment_summary/PaymentSummaryEmail.vue';
+import PaymentSummaryPrivate from '@src/components/shared/payment_summary/PaymentSummaryPrivate.vue';
+import { Country } from '@src/view_models/Country';
+import { Salutation } from '@src/view_models/Salutation';
+import { useI18n } from 'vue-i18n';
 
-const addressTypeComponents = {
-	[ addressTypeName( AddressTypeModel.ANON ) ]: PaymentSummaryAnonymous,
-	[ addressTypeName( AddressTypeModel.COMPANY ) ]: PaymentSummaryCompany,
-	[ addressTypeName( AddressTypeModel.EMAIL ) ]: PaymentSummaryEmail,
-	[ addressTypeName( AddressTypeModel.PERSON ) ]: PaymentSummaryPrivate,
-	[ addressTypeName( AddressTypeModel.UNSET ) ]: PaymentSummaryUnset,
-};
+interface Props {
+	address : Record<string, string>;
+	addressType: String;
+	payment: { interval: any, amount: number, paymentType: any };
+	countries: Array<Country>;
+	languageItem: String;
+	salutations: Array<Salutation>;
+}
+
+const { t, n } = useI18n();
+const props = defineProps<Props>();
+
+const addressTypeComponents = [
+	{ key: addressTypeName( AddressTypeModel.ANON ), component: PaymentSummaryAnonymous },
+	{ key: addressTypeName( AddressTypeModel.COMPANY ), component: PaymentSummaryCompany },
+	{ key: addressTypeName( AddressTypeModel.EMAIL ), component: PaymentSummaryEmail },
+	{ key: addressTypeName( AddressTypeModel.PERSON ), component: PaymentSummaryPrivate },
+	{ key: addressTypeName( AddressTypeModel.UNSET ), component: PaymentSummaryAnonymous },
+];
+
+// TODO: Extract this into a composable to make testing easier
+const addressTypeComponent = addressTypeComponents.find( x => x.key === props.addressType ).component;
+const paymentType = t( props.payment.paymentType );
+const interval = t( 'donation_form_payment_interval_' + props.payment.interval );
+const formattedAmount = n( props.payment.amount, { key: 'currency', currencyDisplay: 'name' } );
+const country = props.countries.find( c => ( c.countryCode === props.address.countryCode || c.countryCode === props.address.country ) ) ?? '';
+const salutation = props.address.salutation ? props.salutations.find( s => s.value === props.address.salutation )?.display : '';
```

Comment:
these probably need to be computed values again, else it only works after reloading the page
```suggestion
const addressTypeComponent = computed( () => addressTypeComponents.find( x => x.key === props.addressType ).component );
const paymentType = computed( () => t( props.payment.paymentType ) );
const interval = computed( () => t( 'donation_form_payment_interval_' + props.payment.interval ) );
const formattedAmount = computed( () => n( props.payment.amount, { key: 'currency', currencyDisplay: 'name' } ) );
const country = computed( () => props.countries.find( c => ( c.countryCode === props.address.countryCode || c.countryCode === props.address.country ) ) ?? '' );
const salutation = computed( () => props.address.salutation ? props.salutations.find( s => s.value === props.address.salutation )?.display : '' );
```

Reply:
created a followup pr https://github.com/wmde/fundraising-app-frontend/pull/181